src/python/WMCore/MicroService/MSRuleCleaner/MSRuleCleaner.py

```diff
@@ -593,15 +593,18 @@ def getMSOutputTransferInfo(self, wflow):
         if transferInfo is None:
             msg = f"Workflow {wflow['RequestName']} is still missing the output transfer document."
             self.logger.warning(msg)
+        elif transferInfo['TransferStatus'] == 'pending':
+            msg = f"Workflow {wflow['RequestName']} is still pending the output data placement."
+            self.logger.warning(msg)
         elif transferInfo['TransferStatus'] == 'done':
             # Set Transfer status - information fetched from MSOutput only
             wflow['TransferDone'] = True
-        elif transferInfo['OutputMap']:
+
             # Set Tape rules status - information fetched from Rucio (tape rule ids from MSOutput)
-            tapeRulesStatusList = []
             # For setting 'TransferTape' = True we require either no tape rules for the
             # workflow have been created or all existing tape rules to be in status 'OK',
             # so every empty TapeRuleID we consider as completed.
+            tapeRulesStatusList = []
```

Comment:
The following is not a request for a change, just a clarification about the difference from the original logic changed with: [#11357:L594-L598](https://github.com/dmwm/WMCore/pull/11357/files#diff-e1c7d744c2a4abc69693fa15853e59e80ef546f3048de9f00a5d30f767c1345cL594-L598)

The way how it was coded initially was allowing an itereation through the `OutputMap` in `MSRuleCleaner` regardless of the transfer status for disk rules. What we end up here now is: we go in this piece only if the `TransferStatus` is done. I hope this is your intention. Otherwise we may run into another trouble.  

And that said, I really need to mention, I had some troubles reading this logic here, because I had to remember the bit that `TransferStatus` in `MSOutput` accounts only for disk rules. We may consider in the future a change of the field name in order to reflect this somehow.

Reply:
Ok, Let me stand myself corrected.

> I had to remember the bit that TransferStatus in MSOutput accounts only for disk rules

This statement I made there is **wrong!** I now read that we have exactly the same logic in `MSOutput` also for tape rules - setting the status initially to `done` and then iterating through the rules in order to figure out if we should revert it to `pending`... and again using the same `TransferStatus` field in the document we upload to the database. 

I do not want to deviate in a discussion, different from the original issue, but just to mention: For clarity and avoiding getting ourselves in confusions similar to the one I've got myself in, I'd suggest we split the `TransferStatus` filed in two - one reflecting the disk rules and another one reflecting the tape ones.

Reply:
Yes, I was mislead by the MSOutput `TransferStatus` parameter as well. The way it's used in the end is:
* the Producer thread creates the MSOutput document, with `TransferStatus=pending`
* once the Consumer thread goes through all the data and creates all the required rules, then it switches to `TransferStatus=done`

So, the check for TransferStatus=pending here is just one extra information for us to know why the check for output tape rules will be skipped (because there won't be any rule ids yet to be looked at).

Reply:
Thanks from confirming this @amaltaro .

As of the `MSOutput` document key `TransferStatus`, I am still in favor of splitting it in two separate ones. But this is definitely something for us to consider eventually in the future and not a topic for discussion in this issue.